testing/adios2/bindings/C/TestBPWriteTypes.cpp

```diff
@@ -116,11 +116,8 @@ TEST_F(BPWriteTypesCC, ADIOS2BPWriteTypes)
         adios2_variable *varR32 = adios2_inquire_variable(ioH, "varR32");
         adios2_variable *varR64 = adios2_inquire_variable(ioH, "varR64");
 
-        size_t size;
-        adios2_variable_name(NULL, &size, varStr);
-        char var_name[size + 1];
-        adios2_variable_name(var_name, &size, varStr);
-        var_name[size] = '\0';
+        const char *var_name;
+        adios2_variable_name(&var_name, varStr);
```

Comment:
@germasch does this even work? `const char*` means can't change var_name contents, but `adios2_variable_name` is modifying them. This is why I advised to stick with the previous solution.

Reply:
If we didn't want to, by convention, return an error code, one would do
```c
const char *name = adios2_variable_name(varStr);
```
The code above does the very same thing, just passing `name` as a by-reference argument. `const` rules in C/C++ are admittedly somewhat confusing, but this is correct. The `const` goes with the `char`.
```c
const char *name = "abc";
name[0] = 'A'; // error
name = "def"; // that's fine, and that' s what adios2_variable_name does, essentially
```
The const is there so that the C code cannot modify the C string, which still lives in std::string's memory. That's why `.c_str()` returns `const char *`.